homeassistant/components/nut/config_flow.py

```diff
@@ -73,40 +100,103 @@ class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
     VERSION = 1
     CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
 
-    async def async_step_user(self, user_input=None):
-        """Handle the initial step."""
+    def __init__(self):
+        """Initialize the nut config flow."""
+        self.nut_config = {}
+        self.available_resources = {}
+        self.ups_list = None
+        self.title = None
+
+    async def async_step_import(self, user_input=None):
+        """Handle the import."""
         errors = {}
         if user_input is not None:
-            if self._host_port_alias_already_configured(
-                user_input[CONF_HOST], user_input[CONF_PORT], user_input.get(CONF_ALIAS)
-            ):
+            if self._host_port_alias_already_configured(user_input):
                 return self.async_abort(reason="already_configured")
-            try:
-                info = await validate_input(self.hass, user_input)
-            except CannotConnect:
-                errors["base"] = "cannot_connect"
-            except Exception:  # pylint: disable=broad-except
-                _LOGGER.exception("Unexpected exception")
-                errors["base"] = "unknown"
+            _, errors = await self._async_validate_or_error(user_input)
+
+            if not errors:
+                title = _format_host_port_alias(user_input)
+                return self.async_create_entry(title=title, data=user_input)
+
+        return self.async_show_form(
+            step_id="user", data_schema=DATA_SCHEMA, errors=errors
+        )
 
-            if "base" not in errors:
-                return self.async_create_entry(title=info["title"], data=user_input)
+    async def async_step_user(self, user_input=None):
+        """Handle the user input."""
+        errors = {}
+        if user_input is not None:
+            info, errors = await self._async_validate_or_error(user_input)
+
+            if not errors:
+                self.nut_config.update(user_input)
+                self.available_resources.update(info["available_resources"])
+                self.ups_list = info["ups_list"]
+
+                if len(self.ups_list) > 1:
+                    # If there is more than one ups
+                    # they need to pick which one so
+                    # we can check which resources are available
+                    return self.async_show_form(
```

Comment:
We should call the step method when we need to switch step. We should not call the show form method.

Reply:
Adjusted, retested, still working as expected